torch/distributed/fsdp/_init_utils.py

```diff
@@ -506,13 +506,19 @@ def _init_prefetching_state(
     return state
 
 
+@no_type_check
 def _init_state_dict_state(state: _FSDPState) -> _FSDPState:
     state._state_dict_type = StateDictType.FULL_STATE_DICT
     state_dict_config: StateDictConfig = FullStateDictConfig()
     state._optim_state_dict_config = FullOptimStateDictConfig()
     state._state_dict_config = state_dict_config
     unshard_params_ctx: Dict[nn.Module, Generator] = {}
     state._unshard_params_ctx = unshard_params_ctx
+
+    if hasattr(state, "_device_mesh"):
+        state._enable_extension = (
+            mesh_resources.get_parent_mesh(state._device_mesh) is not None
+        )
```

Comment:
What if we also let Pippy to use DeviceMesh? Then in Pippy + FSDP case, this should be False but it will be true.

Reply:
> What if we also let Pippy to use DeviceMesh? Then in Pippy + FSDP case, this should be False but it will be true.

Added a TODO to make sure we add additional check once we support different composition.